hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/ScmBlockDeletingServiceMetrics.java

```diff
@@ -196,6 +227,119 @@ public long getNumBlockDeletionTransactionDataNodes() {
+    numBlockDeletionTransactionSent.snapshot(builder, all);
+    numBlockDeletionTransactionSuccess.snapshot(builder, all);
+    numBlockDeletionTransactionFailure.snapshot(builder, all);
+    numBlockDeletionTransactionCompleted.snapshot(builder, all);
+    numBlockDeletionTransactionCreated.snapshot(builder, all);
+    numSkippedTransactions.snapshot(builder, all);
+    numProcessedTransactions.snapshot(builder, all);
+    numBlockDeletionTransactionDataNodes.snapshot(builder, all);
+
+    MetricsRecordBuilder recordBuilder = builder;
+    for (Map.Entry<UUID, DatanodeCommandCounts> e : numCommandsDatanode.entrySet()) {
+      recordBuilder = recordBuilder.endRecord().addRecord(SOURCE_NAME)
+          .add(new MetricsTag(Interns.info("datanode",
+              "Datanode host for deletion commands"), e.getKey().toString()))
+          .addGauge(DatanodeCommandCounts.COMMANDS_SENT_TO_DN,
+              e.getValue().getCommandsSent())
+          .addGauge(DatanodeCommandCounts.COMMANDS_SUCCESSFUL_EXECUTION_BY_DN,
+              e.getValue().getCommandsSuccess())
+          .addGauge(DatanodeCommandCounts.COMMANDS_FAILED_EXECUTION_BY_DN,
+              e.getValue().getCommandsFailure());
+    }
+    recordBuilder.endRecord();
+  }
+
+  /**
+   *  Class contains metrics related to the ScmBlockDeletingService for each datanode.
+   */
+  public static final class DatanodeCommandCounts {
+    private long commandsSent;
+    private long commandsSuccess;
+    private long commandsFailure;
+
+    private static final MetricsInfo COMMANDS_SENT_TO_DN = Interns.info(
+        "CommandsSent",
+        "Number of commands sent from SCM to the datanode for deletion");
+    private static final MetricsInfo COMMANDS_SUCCESSFUL_EXECUTION_BY_DN = Interns.info(
+        "CommandsSuccess",
+        "Number of commands sent from SCM to the datanode for deletion for which execution succeeded.");
+    private static final MetricsInfo COMMANDS_FAILED_EXECUTION_BY_DN = Interns.info(
+        "CommandsFailed",
+        "Number of commands sent from SCM to the datanode for deletion for which execution failed.");
+
+    public DatanodeCommandCounts() {
+      this.commandsSent = 0;
+      this.commandsSuccess = 0;
+      this.commandsFailure = 0;
+    }
+
+    public void incrCommandsSent(long delta) {
+      this.commandsSent += delta;
+    }
+
+    public void incrCommandsSuccess(long delta) {
+      this.commandsSuccess += delta;
+    }
+
+    public void incrCommandsFailure(long delta) {
+      this.commandsFailure += delta;
+    }
+
+    public long getCommandsSent() {
+      return commandsSent;
+    }
+
+    public long getCommandsSuccess() {
+      return commandsSuccess;
+    }
+
+    public long getCommandsFailure() {
+      return commandsFailure;
+    }
+
+    @Override
+    public String toString() {
+      return "Sent=" + commandsSent + ", Success=" + commandsSuccess + ", Failed=" + commandsFailure;
+    }
+  }
+
+  @VisibleForTesting
```

Comment:
nit: I don't think we need this annotation.  It is usually applied to methods that do some internal operation, and are exposed only for tests.  These `getNum...()` methods exists only for tests, which is different.  They don't mutate internal state or do anything "dangerous" that should not be  in production code.  So I think having such safe public methods without adding the annotation is OK.